src/arch/x86/tlb.cc

```diff
@@ -409,7 +418,9 @@ TLB::translate(const RequestPtr &req,
                         entry = insert(alignedVaddr, TlbEntry(
                                 p->pTable->pid(), alignedVaddr, pte->paddr,
                                 pte->flags & EmulationPageTable::Uncacheable,
-                                pte->flags & EmulationPageTable::ReadOnly));
+                                pte->flags & EmulationPageTable::ReadOnly),
+                                        (int)p->pTable->pid());
```

Comment:
Are we using the PID or the hardware TID?

Reply:
I am now trying to use pid everywhere